File: SmartInventoryAPI/Models/Entities/Location.cs
namespace SmartInventoryAPI.Models.Entities;

public class Location
{
    public long ID { get; set; }
    public string? Location_Name { get; set; }
    public string? Outlet_Code { get; set; }
    public int Location_Type { get; set; }
    public string? Address { get; set; }
    public bool Is_Deleted { get; set; } = false;
    public DateTime Creation_Date { get; set; } = DateTime.UtcNow;
}

File: SmartInventoryFrontend/src/types/location.ts
import { Entity } from './common'
import { UserRole } from './user'

export interface Location extends Entity {
  location_Name: string
  outlet_Code: string
  location_Type: number
  address: string
  creation_Date: Date
}

export enum LocationType {
  Outlet = 0,
  Warehouse = 1,
  Distribution_Center = 2,
}

export const LocationTypeLabel: Record<LocationType, string> = {
  [LocationType.Outlet]: 'Outlet',
  [LocationType.Warehouse]: 'Warehouse',
  [LocationType.Distribution_Center]: 'Distribution Center',
}

export interface CreateLocationRequest {
  location_Name: string
  outlet_Code: string
  location_Type: number
  address: string
}

export interface UpdateLocationRequest {
  location_Name?: string
  outlet_Code?: string
  location_Type?: number
  address?: string
}

export interface LocationState {
  locations: Location[]
  currentLocation: Location | null
  loading: boolean
  error: string | null
  total: number
  skip: number
  take: number
}
